sorting/cycle_sort.cpp

```diff
@@ -98,6 +103,21 @@ static void test() {
     std::vector<double> arr2 = sorting::cycle_sort::cycleSort(array2);
     assert(std::is_sorted(std::begin(arr2), std::end(arr2)));
     std::cout << "passed" << std::endl;
+
+    // Test 3
+    // [3, 3, 3, 3] return [3, 3, 3, 3]
+    std::vector<int> array3 = {3, 3, 3, 3};
```

Comment:
Consider using `uint64_t` for non-negative values (or their appropriate size: `uint32_t`, `uint16_t`, `uint8_t`) or `int64_t` for negative values. Check other parts of the code ([reference](https://en.cppreference.com/w/cpp/types/integer)). 🙂

Reply:
Done. Pl see once.